### ISSUE TITLE
Start index for each epoch

### ISSUE THREAD

**Issue by toannguyen1904** [facebookresearch/barlowtwins#44]
Could anyone explain why we do not start at index 0 for every epoch, rather than `epoch * len(loader)`??
Lines 123 to 125:
```
    for epoch in range(start_epoch, args.epochs):
         sampler.set_epoch(epoch)
         for step, ((y1, y2), _) in enumerate(loader, start=epoch * len(loader)):
```

**Comment by jingli9111** [comment]
Hi @toannguyen1904, this is used for the learning rate schedule. The cosine learning rate decay counts how many steps have passed in the total training steps. See lines 155.

**Comment by toannguyen1904** [comment]
> Hi @toannguyen1904, this is used for the learning rate schedule. The cosine learning rate decay counts how many steps have passed in the total training steps. See lines 155.

oh I see it. I just wanna ask one more question. So that for each epoch, because the start index is not 0, we do not train on all data images right?

**Comment by jingli9111** [comment]
Hi @toannguyen1904,
No. It trains on all data each epoch. 
Please just check how `enumerate` works.
